### DISCUSSION TITLE
feat req: Drag & drop zone component

### DISCUSSION THREAD

**Discussion by gitmalong** [saadeghi/daisyui#2315] [Ideas / feature requests]
There is a [file input component](https://daisyui.com/components/file-input/) but no component for displaying a  pretty drag & drop zone which is quite a common element when it comes to uploading files (https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/File_drag_and_drop).

**Comment by saadeghi** [comment]
It's not possible with CSS. It needs JS.  
You can use a [NPM package for that](https://www.npmjs.com/search?q=dropzone) alongside daisyUI.

**Comment by gitmalong** [comment]
I was not thinking about the JS functionality but of a design template for such a drop zone.

**Reply by saadeghi** [reply]
That would be only useful for manually creating a drop zone. Because different drop zone packages and plugins, have their own HTML structure, sometimes their style is customizable, sometimes not. So I'm not sure how is it going to be helpful providing only the visual design of a drop zone when it can't be used with currently available packages and plugins.

**Reply by gitmalong** [reply]
When using the [leptos](https://leptos.dev/) framework such a design element still makes sense as one can simple do:

```rust
// Taken from https://leptos-use.rs/elements/use_drop_zone.html

let drop_zone_el = create_node_ref::<Div>(cx);

let on_drop = |event| {
    // called when files are dropped on zone
};

let UseDropZoneReturn {
    is_over_drop_zone,
    ..
} = use_drop_zone_with_options(
    cx,
    drop_zone_el,
    UseDropZoneOptions::default().on_drop(on_drop)
);

view! { cx,
    <div node_ref=drop_zone_el>
        "Drop files here"
    </div>
}

```

I thought that providing design elements is the core scope of this lib.

**Reply by saadeghi** [reply]
@gitmalong how should different states of dropzone be handled without JS?  
When there's no file, when there's a file being dragged, then file is added, when multiple files are added, when the file type is unallowed, when the file is uploading with a progressbar, when there's a problem in upload...

Do you mean daisyUI should provide different class names for all these states and then you want to swap class names based on the current state of the element? Because that's not something most dropzone plugins would allow us to do.
